src/events/crawler/scrapers/ESP/index.js

```diff
@@ -1,38 +1,149 @@
-          states.push(data);
-        }
-      }
+    const isDate = s => s.includes('/');
+
+    const parseDate = s => {
+      // Source data currently uses ddmmyyyy format for dates,
+      // but is (maybe?) planning to switch to yyyymmdd.
+      // See https://github.com/datadista/datasets/issues/14
+      // When/if this happens, replace contents of this function with:
+      // return LocalDate.parse(s)
+
+      const [d, m, y] = s.split('/');
+      return LocalDate.parse(`${y}-${m}-${d}`);
+    };
+
+    const rawData = {};
+    for (const { name, url } of this.sources) {
+      rawData[name] = await fetch.csv(url, false);
     }
-    states.push(transform.sumData(states));
-    return states;
+
+    // `rawData` looks like this:
+    // ```js
+    // {
+    //    cases: [
+    //      {
+    //        cod_ine: '01',
+    //        CCAA: 'Andalucía',
+    //        '27/02/2020': '1',
+    //        '28/02/2020': '6',
+    //        // ...
+    //      },
+    //      {
+    //        cod_ine: '02',
+    //        CCAA: 'Aragón',
+    //        '27/02/2020': '0',
+    //        '28/02/2020': '1',
+    //        // ...
+    //      },
+    //      //... other locations
+    //    ],
+    //    recovered: [
+    //      // same format
+    //    ],
+    //    deaths: [
+    //      // same format
+    //    ]
+    // }
+    // ```
+
+    const data = rawData.cases
+      .filter(casesRow => casesRow.CCAA !== 'Total')
+      .flatMap(casesRow => {
+        const location = casesRow.CCAA;
+        const isSameLocation = d => d.CCAA === location;
+        const deathsRow = rawData.deaths.find(isSameLocation);
+        const recoveredRow = rawData.recovered.find(isSameLocation);
+        return Object.keys(casesRow)
+          .filter(isDate)
+          .map(date => {
+            return {
+              state: parse.string(location),
+              date: parseDate(date).toString(),
+              cases: parse.number(casesRow[date]),
+              deaths: parse.number(deathsRow[date] || 0),
+              recovered: parse.number(recoveredRow[date] || 0)
+            };
+          });
+      });
+
+    // `data` is an array of objects that look like this:
+    // ```js
+    // {
+    //   location: 'Andalucía',
+    //   date: '2020-02-27',
+    //   cases: 1,
+    //   deaths: 0,
+    //   recovered: 0
+    // },
+    // ```
+
+    // Need to jump through a couple extra hoops here because SCRAPE_DATE doesn't use leading zeroes
+    const scrapeDate = process.env.SCRAPE_DATE
+      ? LocalDate.of(...process.env.SCRAPE_DATE.split('-')) // e.g. '2020-3-8' -> LocalDate.of(2020, 3, 8)
+      : undefined;
+
+    const sampleRow = rawData.cases[0];
+    const dates = Object.keys(sampleRow)
+      .filter(isDate)
+      .map(parseDate)
+      .sort();
+
+    const firstDate = dates[0];
+    const lastDate = dates[dates.length - 1];
+
+    // use the scrape date, or the latest available
+    const queryDate = scrapeDate || lastDate;
```

Comment:
Prefer returning an empty array in case we do not have data for today

Reply:
👍 done